### PR TITLE
Set QueryType correctly in CassandraCodec

### PR DESCRIPTION
**Pull request by Claude-at-Instaclustr**
Fix for #447

### CODE REVIEW

**Comment by rukai** [comment]
Redis cache is currently broken on many levels so its a bit confusing but this is the fix:
image

Firstly the comments were wrong, caching is not done at select time but at insert time. So I reworded them to reflect that.
The second problem is that as the comments say, "SELECTS without a WHERE" currently dont hit the cache.
The result of not hitting the cache is that the correct values are returned whereas when we do hit the cache an empty list is returned.
This is expected because the current implementation of the cache just returns no results.

I'm not sure why your change affected this test but I wouldnt think about it too hard because it changes the tests to better reflect what the results should currently be.

**Review by conorbros — APPROVED** [review]
After you implement @rukai's solution and get the tests passing, it LGTM just some nits.

Also I will change the title of this PR so the commit message is clear.

File: shotover-proxy/src/protocols/cassandra_codec.rs [hunk 1]
```diff
@@ -884,12 +891,15 @@ mod cassandra_protocol_tests {
...
-    fn test_codec_query() {
+    fn test_codec_query_select() {
         let mut codec = new_codec();
         let bytes = hex!(
             "0400000307000000330000002c53454c454354202a2046524f4d20737973
             74656d2e6c6f63616c205748455245206b65793d276c6f63616c27000100"
         );
+
+        // 53454c454354202a2046524f4d2073797374656d2e6c6f63616c205748455245206b65793d276c6f63616c27
+        // 496e7365727420496e746f2073797374656d2e666f6f2028626172292076616c756573202827626172322729
```
[the remark this answers is not in the document]
Comment (Claude-at-Instaclustr) [reply shotover-proxy/src/protocols/cassandra_codec.rs:902]: If you are referring to the commented out code it should be removed.

**Comment by Claude-at-Instaclustr** [comment]
How do I seem to get the simple ones that, like a NASA space telescope, unfold into much bigger problems.

It seems that changing the parser so that it actually sets the QueryType breaks 17 tests -- excluding the one I fixed.

Strangely (or perhaps not strangely) commenting out [line 562 of cassandra_codec.rs](https://github.com/Claude-at-Instaclustr/shotover-proxy/blob/669b6ea20be2d9e459ad883f4c440f45dff8c9ae/shotover-proxy/src/protocols/cassandra_codec.rs#L562) makes all the problems go away.  So all the problem are tied to the fact that the MessageDetails.query_type  is going from QueryType::Read  to QueryType::Write.

Much of this seem to be redis cache errors which certainly brings home @rukai  statement: "Redis cache is currently broken on many levels"

**Comment by rukai** [comment]
I checked out your branch and reenabled the redis-cache test fix and ran `cargo test`.
A bunch of redis tests failed because at some point a redis container failed to shutdown. Shutting it down allowed these tests to pass.

I then reran `cargo test` and every test passed.
So maybe its a test order or timing issue?

**Comment by Claude-at-Instaclustr** [comment]
@rukai any idea how to fix a test ordering issue?  Why would there be a test ordering issue?  Is this perhaps a case where parallel tests are interfering with each other?

**Comment by rukai** [comment]
this time it failed a rust fmt check

**Comment by rukai** [comment]
The main branch post-checkin ci run failed so I ran it again and then it failed again. https://github.com/shotover/shotover-proxy/runs/4837120513?check_suite_focus=true

So some intermittent failures might have made their way onto main branch and that might be what your experiencing.

**Comment by Claude-at-Instaclustr** [comment]
What do we do to fix this?

**Comment by rukai** [comment]
run `cargo fmt`

**Comment by Claude-at-Instaclustr** [comment]
`cargo fmt` makes no change to the files.

**Comment by rukai** [comment]
It works as expected on my machine:
image

**Comment by rukai** [comment]
Have you maybe already comitted the changes from `cargo fmt` but forgotten to push?

**Review by rukai — APPROVED** [review]
lgtm

**Pull request merged**